src/microbe_stage/editor/MicrobePartSelection.cs

```diff
@@ -110,6 +125,14 @@ public override void _Ready()
         UpdateIcon();
     }
 
+    public override void _Process(float delta)
+    {
+        base._Process(delta);
+
+        if (nameLabel != null)
+            nameLabel.Visible = Settings.Instance.DisplayPartNames || AlwaysShowLabel;
```

Comment:
This should not use _Process, this will unnecessarily sap performance. Use the listener for option changes register to detect changes to the option.